doc/Build_Environment.rst

```diff
@@ -0,0 +1,348 @@
+|with-db4=db4-path       |    Db4 location we want to use for the metadata server.                                   |
++------------------------+-------------------------------------------------------------------------------------------+
+|with-linux=linux-path   | Linux kernel location we want to use for kernel components                                |
++------------------------+-------------------------------------------------------------------------------------------+
+|with-gss=gss-path       | Gss libraries location we want to use for security.                                       |
++------------------------+-------------------------------------------------------------------------------------------+
+|build-dir=build-path    |  Location where the object files and binaries should be generated.                        |
++------------------------+-------------------------------------------------------------------------------------------+
+
+Generation Part
+===============
+
+Here is the list if build part functionality:
+
+- Options provided by configure part are taken into account, including paths, debug and optimization options;
+
+- Support for simultaneous builds and generating binaries out of the source tree;
+
+- Support for features like generating source code from templates (db4 fol records).
+
+Repository Part
+===============
+
+The following functionality is provided:
+
+- Source code is stored in a repository, supporting branches and commit hooks;
+
+- Commit hooks are set to mail repository changes to the peers. Commit emails contain standard diff of the changes maid.
+
+*********************
+Logical Specification
+*********************
+
+Configuration Part
+===================
+
+Configuration part, in whatever build system it is implemented, check the following system specifics:
+
+- Platform check - part of standard check in Autotools and Cmake. Very easy to figure out in Scons;
+
+- Build tools, such as compiler, linker, etc., are usualy the part of standard configure checks for all build systems;
+
+- Header files check - one line macro/function in all build systems;
+
+- Libraries checks - one line macro/function in all build systems;
+
+- Not standard functions - one line macro/function in all build systems;
+
+- Configure options - handled differently in all build systems but still is very easy to talk much about. DLD will show examples of all the checks including configure options checks.
+
+Generation Path
+===============
+
+Generation part is very simplistic for all of the build systems described. One just specifies what targets needed: binaries, libraries, etc. The only exception is that, we may possibly need handling for not standard source files, such as fol template, to generate fol tables for db4.
+
+Repository Path
+================
+
+So far there are three repositories for storing different logical parts of the Motr source tree:
+
+.. image:: Images/REP.PNG
```

Comment:
Remove this image.
Use table, please.

The same as the above one.

Reply:
Incorporated